src/python_testing/TC_MCORE_FS_1_1.py

```diff
@@ -0,0 +1,150 @@
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+
+import logging
+import os
+import pathlib
+import random
+import signal
+import subprocess
+import time
+import uuid
+
+import chip.clusters as Clusters
+from chip import ChipDeviceCtrl
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_MCORE_FS_1_1(MatterBaseTest):
+
+    @async_test_body
+    async def setup_class(self):
+        super().setup_class()
+        # TODO: This needs to come from an arg and needs to be something available on the TH
+        # TODO: confirm whether we can open processes like this on the TH
+        # TODO: When bridge device is more stable switch from acll-clusters-app to bridge app.
+        app = os.path.join(pathlib.Path(__file__).resolve().parent, '..', '..', 'out',
+                           'linux-x64-all-clusters-no-ble', 'chip-all-clusters-app')
+
+        self.kvs = f'kvs_{str(uuid.uuid4())}'
+        self.port = 5543
+        discriminator = random.randint(0, 4095)
+        discriminator = 3840
+        passcode = 20202021
+        app_args = f'--secured-device-port {self.port} --discriminator {discriminator} --passcode {passcode} --KVS {self.kvs}'
+        cmd = f'{app} {app_args}'
+        # TODO: Determine if we want these logs cooked or pushed to somewhere else
+        logging.info("Starting application to acts mock a server portion of TH_FSA")
+        self.app_process = subprocess.Popen(cmd, bufsize=0, shell=True)
+        logging.info("Started application to acts mock a server portion of TH_FSA")
+        time.sleep(3)
+
+        logging.info("Commissioning from separate fabric")
+        # Create a second controller on a new fabric to communicate to the server
+        new_certificate_authority = self.certificate_authority_manager.NewCertificateAuthority()
+        new_fabric_admin = new_certificate_authority.NewFabricAdmin(vendorId=0xFFF1, fabricId=2)
+        paa_path = str(self.matter_test_config.paa_trust_store_path)
+        self.TH_server_controller = new_fabric_admin.NewController(nodeId=112233, paaTrustStorePath=paa_path)
+        self.server_nodeid = 1111
+        await self.TH_server_controller.CommissionOnNetwork(nodeId=self.server_nodeid, setupPinCode=passcode, filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=discriminator)
+        logging.info("Commissioning TH_SERVER complete")
+
+    def teardown_class(self):
+        logging.warning("Stopping app with SIGTERM")
+        self.app_process.send_signal(signal.SIGTERM.value)
+        self.app_process.wait()
+        # TODO: Use timeout, if term doesn't work, try SIGINT
```

Comment:
is this still under development? can this be cleaned up if it just works?

Reply:
again, just copied from my code. I'll add a cleanup PR, but it would be better to consolidate these if you're using the same setup.

It does work, but for some reason we use sigint in the test runner. Which, IMO is incorrect.